app/src/main/java/com/github/epfl/meili/profile/favoritepois/FavoritePoisActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.epfl.meili.profile.favoritepois
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.github.epfl.meili.R
+import com.github.epfl.meili.database.FirestoreDatabase
+import com.github.epfl.meili.forum.ForumActivity
+import com.github.epfl.meili.home.Auth
+import com.github.epfl.meili.map.MapActivity
+import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.models.FavoritePointOfInterest
+import com.github.epfl.meili.util.MeiliViewModel
+import com.github.epfl.meili.util.TopSpacingItemDecoration
+import java.util.*
+
+
+class FavoritePoisActivity : AppCompatActivity() {
+    companion object {
+        private const val CARD_PADDING: Int = 30
+        private const val ACTIVITY_TITLE = "Favorite POIs"
+        private const val DB_PATH = "poi-favorite/%s/poi-favorite"
+
+        fun addPoiToFavorites(userKey: String, poi: PointOfInterest) {
+            FirestoreDatabase( // add to poi favorites
+                String.format(DB_PATH, userKey),
+                FavoritePointOfInterest::class.java
+            ).addElement(poi.uid, FavoritePointOfInterest(poi))
```

Comment:
This creates a FirestoreDatabase every time a POI is added, which is not great IMO. Instead, the database should be instantiated in the view model of the activity where the database is used (here, Forum).

Reply:
Good catch! I moved the db to the forum's view model. But I did keep this method just for testing purposes.

Reply:
Thanks for changing this! The method seems redundant to me still, why can't we test the db inside the forum's view model?